### PR TITLE
kadm5.acl code refactoring

### PR DESCRIPTION
**Pull request by greghudson**
These commits move and refactor the kadm5.acl code in preparation for adding the kadm5_auth interface.  There is a minor functional difference in the third commit, so I split it out and assigned it a ticket number.

The second commit doesn't leave much of the original code intact, so the changes to auth_acl.c will be difficult to review as a diff.  It's not a lot of code, so reviewing the refactored code as if it were new is probably the best approach.

### CODE REVIEW

File: src/kadmin/server/auth_acl.c [hunk 1]
```diff
@@ -72,7 +66,6 @@ struct wildstate {
 
 struct acl_state {
```
Comment (frozencemetery) [inline src/kadmin/server/auth_acl.c]: This struct only has a single element - is it worth keeping the type?

File: src/kadmin/server/auth_acl.c [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+};
+
+struct acl_state {
+    struct acl_entry *list;
+};
+
+static struct acl_state aclstate;
+
+/*
+ * Get a line from the ACL file into buf, with maximum size bufsize.  Lines
```
Comment (frozencemetery) [inline src/kadmin/server/auth_acl.c]: There's no parameter bufsize around; I think this is referring to chunksize, which is hardcoded to 128?

File: src/kadmin/server/auth_acl.c [hunk 3]
```diff
@@ -0,0 +1,588 @@
...
+    { 'e', ACL_EXTRACT },
+    { '\0', 0 }
+};
+
+struct wildstate {
+    int nwild;
+    const krb5_data *backref[9];
+};
+
+struct acl_state {
```
Comment (frozencemetery) [inline src/kadmin/server/auth_acl.c]: Is this worth keeping around as a type?  It's a struct with only one element, and not used elsewhere.

File: src/kadmin/server/auth_acl.c [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+    struct acl_entry *list;
+};
+
+static struct acl_state aclstate;
+
+/*
+ * Get a line from the ACL file into buf, with maximum size bufsize.  Lines
+ * ending with \ are continued on the next line.  The caller should set *lineno
+ * to 1 and *incr to 0 before the first call.  On successful return, *lineno
+ * will be the line number of the line read.  Return true on success, false on
```
Comment (frozencemetery) [inline src/kadmin/server/auth_acl.c]: Function does not return true/false; instead, it seems to return NULL on failure.

File: src/kadmin/server/auth_acl.c (hunk 3 above)
Comment (greghudson) [inline src/kadmin/server/auth_acl.c]: struct acl_state will become the module data type (and we will stop using any global state in this file) when auth_acl.c becomes a kadm5_auth handle.  We could use struct acl_entry ** as the module data type instead, but that is harder to think about.

File: src/kadmin/server/auth_acl.c (hunk 2 above)
Comment (greghudson) [inline src/kadmin/server/auth_acl.c]: A relic of an earlier rewrite of that function.  I will fix it.  chunksize is not visible to the caller.

File: src/kadmin/server/auth_acl.c (hunk 4 above)
Comment (greghudson) [inline src/kadmin/server/auth_acl.c]: Also a relic of an earlier rewrite.

**Pull request merged**